react/utils/normalize.ts

```diff
@@ -1,10 +1,20 @@
+import type { ProductTypes } from 'vtex.product-context'
```

Comment:
This file has become way more complex than before, no problem with that. But could you write some tests for these new functions?

Since they are simple TS functions, you wouldn't need to use anything besides Jest to do so :)

Reply:
I'm not familiar with it, but I'll see what I can do